src/main/java/com/mraof/minestuck/alchemy/GristHelper.java

```diff
@@ -135,58 +141,61 @@ public static void increase(Level level, PlayerIdentifier player, GristSet set)
 		data.setGristCache(newCache);
 	}
 	
-	public static void notify(MinecraftServer server, PlayerIdentifier player, GristSet set)
+	public static void notify(MinecraftServer server, PlayerIdentifier player, GristSet set, GristToast.EnumSource source, boolean increase)
 	{
 		if(MinestuckConfig.SERVER.showGristChanges.get())
 		{
-			Map<GristType, Long> reqs = set.getMap();
-			for(Entry<GristType, Long> pairs : reqs.entrySet())
+			
+			//don't know why this is here, it was copied over from when i merged notifyEditMode() and notify(), but im not sure what it does/did. Seems to be a sanity check.
+			SburbConnection sc;
+			EditData ed;
+			
+			if(source == GristToast.EnumSource.SERVER) {
+				sc = SkaianetHandler.get(server).getActiveConnection(player);
+				if(sc == null)
+					return;
+				
+				ed = ServerEditHandler.getData(server, sc);
+				if(ed == null)
+					return;
+			}
+			
+			if(player.getPlayer(server).isLocalPlayer() == true)
```

Comment:
This should always be false when you're getting the player from a server via a player identifier.

Reply:
Ah, okay. It's been removed.